packages/@aws-cdk/aws-apigateway/test/test.integration.ts

```diff
@@ -33,6 +34,89 @@ export = {
     test.done();
   },
 
+  'uri is self determined from the NLB'(test: Test) {
+    const stack = new cdk.Stack();
+    const vpc = new ec2.Vpc(stack, 'VPC');
+    const nlb = new elbv2.NetworkLoadBalancer(stack, 'NLB', { vpc });
+    const link = new apigw.VpcLink(stack, 'link', {
+      targets: [nlb],
+    });
+    const api = new apigw.RestApi(stack, 'restapi');
+    const integration = new apigw.Integration({
+      type: apigw.IntegrationType.HTTP_PROXY,
+      integrationHttpMethod: 'ANY',
+      options: {
+        connectionType: apigw.ConnectionType.VPC_LINK,
+        vpcLink: link,
+      },
+    });
+    api.root.addMethod('GET', integration);
+
+    expect(stack).to(haveResourceLike('AWS::ApiGateway::Method', {
+      Integration: {
+        Uri: {
+          'Fn::Join': [
+            '',
+            [
+              'http://',
+              {
+                'Fn::GetAtt': [
+                  'NLB55158F82',
+                  'DNSName',
+                ],
+              },
+            ],
+          ],
+        },
+      },
+    }));
+
+    test.done();
+  },
+
+  'uri must be set for VpcLink with multiple NLBs'(test: Test) {
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app);
+    const vpc = new ec2.Vpc(stack, 'VPC');
+    const nlb1 = new elbv2.NetworkLoadBalancer(stack, 'NLB1', { vpc });
+    const nlb2 = new elbv2.NetworkLoadBalancer(stack, 'NLB2', { vpc });
+    const link = new apigw.VpcLink(stack, 'link', {
+      targets: [nlb1, nlb2],
+    });
+    const api = new apigw.RestApi(stack, 'restapi');
+    const integration = new apigw.Integration({
+      type: apigw.IntegrationType.HTTP_PROXY,
+      integrationHttpMethod: 'ANY',
+      options: {
+        connectionType: apigw.ConnectionType.VPC_LINK,
```

Comment:
If I specify `vpclink` why do I also need to specify connection type? Seems redundant